.github/workflows/lint-test.yaml

```diff
@@ -72,9 +72,21 @@ jobs:
           kubectl_version: ${{ inputs.kind_kubectl_version }}
           node_image: ${{ inputs.kind_node_image }}
 
-      - name: Apply Gateway API CRDs
+      - name: Check Kubernetes version and apply Gateway API CRDs
         if: steps.list-changed.outputs.changed == 'true'
```

Comment:
As an alternative, could something like that work

```
if: inputs.kind_kubectl_version > "1.23"
```

Reply:
technically it's not the kubectl version that controls whether we can accept the CRD, but the server version. I'm not even super sure why we have two, but we do.

Also the only way i could find to compare semver in GH actions is to use a subaction, which looks like an overkill for two lines of bash.